src/Npgsql/PostgresEnvironment.cs

```diff
@@ -55,4 +79,30 @@ static class PostgresEnvironment
         => GetHomeDir() is string homedir
             ? Path.Combine(homedir, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "postgresql" : ".postgresql")
             : null;
-}
\ No newline at end of file
+
+    static string? GetSystemPostgresDir() => Environment.GetEnvironmentVariable("PGSYSCONFDIR");
+
+    static string? GetDefaultSystemPostgresDir() => ExecuteCommandSync("pg_config", "--sysconfdir");
+
+    static string? ExecuteCommandSync(string command, string arguments)
```

Comment:
Since this already contains 'Sync' and uses APIs that can block I guess that you also plan an Async variant (also for `GetConfigFromIniFile()`) - no?
The internal part could use a boolean async flag like we do it all the time in Npgsql (e. g. see [KerberosUsernameProvider](https://github.com/npgsql/npgsql/blob/main/src/Npgsql/KerberosUsernameProvider.cs) for how to do this for `Process`)

Reply:
Done.
I have shared the code  between both usages.